newsroom/template_filters.py

```diff
@@ -61,6 +70,113 @@ def parse_date(datetime):
     return datetime
 
 
+def get_schedule_type(start: datetime, end: datetime, all_day: bool, no_end_time: bool) -> ScheduleType:
+    """
+    Determine the schedule type based on event start and end times.
+
+    params :
+    - start (datetime): Start time of the event.
+    - end (datetime): End time of the event.
+    - all_day (bool): True if the event is an all-day event.
+    - no_end_time (bool): True if the event has no specified end time.
+
+    return:
+    ScheduleType: The type of schedule (ALL_DAY, NO_DURATION, MULTI_DAY, REGULAR).
+    """
+    duration = int((end - start).total_seconds() / 60) if not no_end_time and end else 0
+    DAY_IN_MINUTES = 24 * 60 - 1
+
+    if all_day:
+        return ScheduleType.ALL_DAY if duration in (0, DAY_IN_MINUTES) else ScheduleType.MULTI_DAY
+
+    if no_end_time:
+        return ScheduleType.NO_DURATION
+
+    if duration >= DAY_IN_MINUTES and start.date() != (end.date() if end else None):
+        return ScheduleType.MULTI_DAY
+
+    if duration <= DAY_IN_MINUTES and start.date() == (end.date() if end else None):
+        return ScheduleType.ALL_DAY
+
+    if duration == 0 and start.time() == (end.time() if end else None):
+        return ScheduleType.NO_DURATION
+
+    return ScheduleType.REGULAR
+
+
+def is_item_tbc(item: dict) -> bool:
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    planning = item.get("planning_items", [])
+    return event_tbc or (planning and planning[0].get("_time_to_be_confirmed", False))
+
+
+def format_event_datetime(item: dict) -> str:
+    date_info = item.get("dates", {})
+
+    if not date_info:
+        return ""
+
+    tz = date_info.get("tz", get_session_timezone())
+    try:
+        # Set the session timezone
+        set_session_timezone(tz)
+
+        start = parse_date(date_info.get("start"))
+        end = parse_date(date_info.get("end")) if date_info.get("end") else None
+        all_day = date_info.get("all_day")
+        no_end_time = date_info.get("no_end_time")
+        is_tbc_item = is_item_tbc(item)
+
+        schedule_type = get_schedule_type(start, end, all_day, no_end_time)
+
+        formatted_start = datetime_long(start)
+        formatted_end = datetime_long(end) if end else None
+
+        if is_tbc_item:
+            if end and start.date() != end.date():
+                return lazy_gettext("Event Starts: {start} - Event Ends: {end} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, end=formatted_end, tz=tz
+                )
+
+            else:
+                return lazy_gettext("Event Starts: {start} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, tz=tz
+                )
+
+        if (end and start.date() != end.date()) and schedule_type in (ScheduleType.REGULAR, ScheduleType.MULTI_DAY):
```

Comment:
if you use the schedule enum imo it should only use that after to figure out how to present it, but using again the end and start here is imo wrong.
would be good also to check that all schedule [cases are covered](https://typing.readthedocs.io/en/latest/source/unreachable.html#assert-never-and-exhaustiveness-checking)

Reply:
got it